driver/reduceextreme_driver.hpp

```diff
@@ -0,0 +1,509 @@
+    {
+        STOP_TIME
+        int32_t iter = inflags.GetValueInt("iter");
+        if(WALL_CLOCK)
+            std::cout << "Wall-clock Time Forward ReduceExtreme Elapsed: " << t.gettime_ms() / iter
+                      << " ms\n";
+
+        float kernel_average_time =
+            iter > 1 ? (kernel_total_time - kernel_first_time) / (iter - 1) : kernel_first_time;
+        std::cout << "GPU Kernel Time Forward ReduceExtreme Elapsed: " << kernel_average_time
+                  << " ms\n";
+    }
+
+    if(indice_dev->FromGPU(GetStream(), indice.data()) != 0)
+        std::cerr << "Error copying (indice_dev) from GPU, size: " << indice_dev->GetSize()
+                  << std::endl;
+
+    if((reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MIN) ||
+       (reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MAX))
+    {
+        if(y_dev->FromGPU(GetStream(), y.data()) != 0)
+            std::cerr << "Error copying (y_dev) from GPU, size: " << y_dev->GetSize() << std::endl;
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ReduceExtremeDriver<Tgpu, Tref>::RunForwardCPU()
+{
+    if(reduceExtremeOp == MIOPEN_REDUCE_EXTREME_ARGMIN)
+    {
+        mloReduceExtremeForwardRunHost<Tgpu, Tref, ReduceExtremeOp_t::Argmin>(
+            xDesc, nullptr, indiceDesc, x.data(), nullptr, indicehost.data(), dim);
+    }
+    else if(reduceExtremeOp == MIOPEN_REDUCE_EXTREME_ARGMAX)
+    {
+        mloReduceExtremeForwardRunHost<Tgpu, Tref, ReduceExtremeOp_t::Argmax>(
+            xDesc, nullptr, indiceDesc, x.data(), nullptr, indicehost.data(), dim);
+    }
+    else if(reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MIN)
+    {
+        mloReduceExtremeForwardRunHost<Tgpu, Tref, ReduceExtremeOp_t::Min>(
+            xDesc, yDesc, indiceDesc, x.data(), yhost.data(), indicehost.data(), dim);
+    }
+    else if(reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MAX)
+    {
+        mloReduceExtremeForwardRunHost<Tgpu, Tref, ReduceExtremeOp_t::Max>(
+            xDesc, yDesc, indiceDesc, x.data(), yhost.data(), indicehost.data(), dim);
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ReduceExtremeDriver<Tgpu, Tref>::RunBackwardGPU()
+{
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ReduceExtremeDriver<Tgpu, Tref>::VerifyForward()
+{
+    RunForwardCPU();
+
+    if((reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MIN) ||
+       (reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MAX))
+    {
+        auto error = miopen::rms_range(yhost, y);
+
+        if(!std::isfinite(error) || std::abs(static_cast<float>(error)) != 0.0f)
+        {
+            std::cout << "Forward ReduceExtreme FAILED: Result does not equal" << std::endl;
+            return EC_VerifyFwd;
+        }
+        else
+        {
+            std::cout << "Forward ReduceExtreme Verifies on CPU and GPU (err=" << error << ")\n";
+        }
+    }
+    auto indice_error = miopen::rms_range(indicehost, indice);
```

Comment:
Tensors should be exactly the same, isn't it?

Reply:
The error check above was the problem. fixed it.

Reply:
@seungmanhan 
> Tensors should be exactly the same, isn't it?

But am I right?

If yes, then: another Q: does `rms_range` guarantee that its return value is 0 only when contents of buffers are identical? I am in doubt...

Reply:
change rms_range to mismatch_idx function.

Reply:
[resolved]